### DISCUSSION TITLE
Bored with magic tricks?

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#23275] [Quarkus Blog/Website]
# Bored with magic tricks?

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/magic-control/

**Comment by vukzeka** [comment]
Hi Clement,
Thanks for such a great article. Really helpful to understand quarkus connection with Vertx and underlying "magic".

I have one question, is it feasible to use managed HTTP server or Regular Quarkus but start underlying Vertx/HTTP server on demand/programmatically?
If we add quarkus-vertx-http as dependency in quarkus app that will automatically start HTTP server and currently is required if we want to create custom Router eg. to achieve something like below

```
@Dependent
public class HttpVerticle extends AbstractVerticle {

  @Override
  public void start() {
    HttpServer server = vertx.createHttpServer();
    Router router = Router.router(vertx);
    router.route().handler(routingContext -> {
      HttpServerResponse response = routingContext.response();
      response.putHeader("content-type", "text/plain");
      response.end("Hello world");
    });
    server.requestHandler(router).listen(8080);
  }
}
```


This would be useful for command mode applications where is required to start server, do some activity, and then stop it after.
I know that this can be achieved by using Vertx as direct dependency but keen to understand if something like that can be achieved with managed approach?

Thanks.

**Comment by tbkahuna48** [comment]
Nicely done Clement
